models/job_models.go

```diff
@@ -0,0 +1,81 @@
+package models
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+//go:generate reform
+
+// EchoJobResult stores echo job specific result data.
+type EchoJobResult struct {
+	Message string `json:"message"`
+}
+
+// JobType represents job type.
+type JobType string
+
+// Supported job types.
+const (
+	Echo = JobType("echo")
+)
+
+// JobResult describes a job result which is storing in persistent storage.
+//reform:job_results
+type JobResult struct {
+	ID         string    `reform:"id,pk"`
+	PMMAgentID string    `reform:"pmm_agent_id"`
+	Type       JobType   `reform:"type"`
+	Done       bool      `reform:"done"`
+	Error      string    `reform:"error"`
+	Result     []byte    `reform:"result"`
```

Comment:
also please check backup_model.go, you can see that we marshal and unmarshal this kind of fields the other way. I'm almost sure that we can reuse that.

Reply:
Ther is no such file `backup_model.go`, you mean to create a column per job type?
```
	EchoJobResult      *EchoJobResult      `reform:"echo_job_result"`
	SomeJobResult     *SomeJobResult     `reform:"some_job_result"`
	AnotherJobResult  *AnotherJobResult `reform:"another_job_result"`
````
Well, it can be done that way, but it will require migration each time when a new job introduced.

Reply:
Done